docs/3_finetuner_in_action/3_3_text_to_image_using_clip.md

```diff
@@ -1,3 +1,107 @@
 # Text to image search using CLIP
 
-Coming soon!
\ No newline at end of file
+This guide will showcase fine-tuning a `CLIP` model for text to image retrieval.
+
+## Task overview
+We'll be fine-tuning CLIP on a fashion captioning dataset which contains information about fashion products.
+
+For each product the dataset contains a title and images of multiple variants of the product. We constructed a parent `Document` for each picture, which contains two chunks: an image document and a text document holding the description of the product.
+
+
+## Preparing data
+Training and evaluation data are already prepared and pushed to Hubble following the [instructions](../2_step_by_step/2_4_create_training_data.md).
+You can either pull the data:
+```python
+from docarray import DocumentArray
+train_data = DocumentArray.pull('clip-fashion-train-data')
+eval_data = DocumentArray.pull('clip-fashion-eval-data')
+```
+Or specify given `DocumentArray` names (`clip-fashion-train-data` and `clip-fashion-eval-data`) directly to finetuner.
+
+
+## Choosing the model
+Currently, we only support `openai/clip-vit-base-patch32` for text to image retrieval tasks. However, you can see all available models either in [the docs](../2_step_by_step/2_5_choose_back_bone.md) or by calling `finetuner.describe_models()`.
+
+
+## Creating a fine-tuning job
+Now that everything's ready, let's create a fine-tuning run!
+
+```{admonition} Login to Jina Cloud
+:class: tip
+Before creating a run, you need to [Login to Jina ecosystem](../2_step_by_step/2_3_login_to_jina_ecosystem.md) by calling `finetuner.login()`.
+```
+
+```python
+from finetuner.callback import BestModelCheckpoint, EvaluationCallback
+
+run = finetuner.fit(
+        model='openai/clip-vit-base-patch32',
+        run_name='clip-fashion',
+        train_data='clip-fashion-train-data',
+        eval_data='clip-fashion-eval-data',
+        epochs=5,
+        learning_rate= 1e-5,
+        callbacks=[BestModelCheckpoint(), EvaluationCallback(query_data='clip-fashion-eval-data')],
+        loss='CLIPLoss',
+        image_modality='image',
+        text_modality='text',
```

Comment:
can we add some sentence below to explain the relationship between these two arguments and how we create DocumentArray?